scrapy/downloadermiddlewares/redirect.py

```diff
@@ -98,6 +116,7 @@ def process_response(self, request, response, spider):
         if "Location" not in response.headers or response.status not in allowed_status:
             return response
 
+        assert response.headers["Location"]
```

Comment:
Would mypy understand this?

```suggestion
        assert "Location" in response.headers
```

or

```suggestion
        assert isinstance(response.headers["Location"], bytes)
```

Otherwise, we are saying that a Location header cannot be an empty string, which I imagine it technically can.